bin/check_fasta_insdc.py

```diff
@@ -64,29 +69,109 @@ def get_refget_metadata(md5_digest):
+                results.add(contigEntity['assembly']['insdcAccession'])
+        if '_links' in response_data and 'next' in response_data['_links']:
+            # Add results from next page if needed
+            results |= _get_containing_assemblies_paged(response_data['_links']['next'])
+        return results
+    return set()
+
+
+def get_containing_assemblies(md5_digest):
+    # Wrapper method to handle pagination
+    url = f'{CONTIG_ALIAS_SERVER}/{md5_digest}'
+    return _get_containing_assemblies_paged(url)
+
+
+def assess_fasta(input_fasta, analyses, metadata_insdc):
+    """
+    Check whether all sequences in fasta file are INSDC, and if so whether the INSDC accession provided in the metadata
+    is compatible.
+    :param input_fasta: path to fasta file
+    :param analyses: aliases of all analyses associated with this fasta (used only for reporting)
+    :param metadata_insdc: INSDC accession from metadata (if None will only do the first check)
+    :returns: dict of results
+    """
     results = {'sequences': []}
     all_insdc = True
+    possible_assemblies = set()
     for header, sequence in fasta_iter(input_fasta):
         name = header.split()[0]
         md5_digest = refget_md5_digest(sequence)
         sequence_metadata = get_refget_metadata(md5_digest)
-        results['sequences'].append({'sequence_name': name, 'sequence_md5': md5_digest, 'insdc': bool(sequence_metadata)})
-        all_insdc = all_insdc and bool(sequence_metadata)
+        is_insdc = bool(sequence_metadata)
+        if is_insdc:
+            containing_assemblies = get_containing_assemblies(md5_digest)
+            if len(containing_assemblies) == 0:
+                logger.warning(f'Sequence with this MD5 is INSDC but not found in contig alias: {md5_digest}')
+                continue
+            if len(possible_assemblies) == 0:
+                possible_assemblies = containing_assemblies
+            else:
+                possible_assemblies &= containing_assemblies
+        results['sequences'].append({'sequence_name': name, 'sequence_md5': md5_digest, 'insdc': is_insdc})
+        all_insdc = all_insdc and is_insdc
     results['all_insdc'] = all_insdc
+
+    # Only report on metadata concordance if all of the following hold:
+    #  1) All sequences in FASTA file are INSDC
+    #  2) At least one compatible assembly accession was found in contig alias
+    #  3) Found a single assembly accession in the metadata to compare against this FASTA file
+    # If (3) is missing but (1) and (2) hold, we will still report possible INSDC assemblies.
+    if all_insdc and possible_assemblies:
+        results['possible_assemblies'] = possible_assemblies
+    if all_insdc and possible_assemblies and metadata_insdc:
+        results['metadata_assembly_compatible'] = (metadata_insdc in possible_assemblies)
+        results['associated_analyses'] = analyses
+        results['assembly_in_metadata'] = metadata_insdc
```

Comment:
We could debate this but in my view we would want `check_fasta_insdc` to report as much as possible and filter what needs to be reported in the final report.
 - I would report all possible assemblies if any was found.
 - `metadata_assembly_compatible` per analysis `if bool(metadata_insdc) and bool(possible_assemblies)`
 I think I'm mostly worried about the case where the assembly fasta contains one non-insdc sequence and all the check fails.

Reply:
I think that makes sense, right now it's kind of assuming that the is_insdc check and contig alias checks are always 100% accurate and consistent with each other, which is unlikely... I'll update this.

Reply:
The new implementation would allow a report like this:
![image](https://github.com/EBIvariation/eva-sub-cli/assets/5599307/3373eebc-4c20-4f60-b588-0a177fd392d4)
For a case where e.g. a sequence is not supported by the refget endpoint, but according to contig alias it's consistent with the metadata. It will also show two errors if not everything is INSDC and the metadata is not consistent.

The only case where the second check doesn't show up at all in the final report is if we don't find any possible assemblies - I think I was thinking that we're expecting contig alias might be missing things and so we wouldn't want to output an error, but perhaps we should show it as an error anyway so both checks always show up? What do you think?

Reply:
I agree that their will be a number of cases where the contig alias will not have the assembly. We should not show the error in that case. 
We probably will come back to this decision if we fell like the contig alias has enough coverage of assemblies we're using. But for the moment this is sensible.